### PR TITLE
fix(middleware-signing): attempt secondary authscheme selection during request signing

### PR DESCRIPTION
**Pull request by kuhe**
### Issue
n/a

### Description
In awsAuthMiddleware, select the second authScheme if:
- second auth scheme exists
- first auth scheme is sigv4a
- sigv4a is not available

### Testing
existing unit test

- [x] e2e
- [x] e2e:legacy

### CODE REVIEW

File: packages/middleware-signing/src/awsAuthConfiguration.ts [hunk 1]
```diff
@@ -196,12 +204,18 @@ export const resolveAwsAuthConfig = <T>(
...
-      input.signingRegion = input.signingRegion || signingRegion;
-      // signing name resolving order:
-      // user supplied signingName -> endpoints.json inferred (credential scope -> model arnNamespace) -> model service id
-      input.signingName = input.signingName || signingService || input.serviceId;
+
+      if (overwrite) {
+        input.signingRegion = originalInputSigningRegion || signingRegion;
+        input.signingName = originalInputSigningName || signingService || input.serviceId;
+      } else {
+        // update client's singing region and signing service config if they are resolved.
```
Comment (syall) [inline packages/middleware-signing/src/awsAuthConfiguration.ts:212]:
```suggestion
        // update client's signing region and signing service config if they are resolved.
```

File: packages/middleware-signing/src/awsAuthConfiguration.ts [hunk 2]
```diff
@@ -196,12 +204,18 @@ export const resolveAwsAuthConfig = <T>(
...
       const signingRegion = authScheme.signingRegion;
       const signingService = authScheme.signingName;
-      // update client's singing region and signing service config if they are resolved.
-      // signing region resolving order: user supplied signingRegion -> endpoints.json inferred region -> client region
-      input.signingRegion = input.signingRegion || signingRegion;
-      // signing name resolving order:
-      // user supplied signingName -> endpoints.json inferred (credential scope -> model arnNamespace) -> model service id
-      input.signingName = input.signingName || signingService || input.serviceId;
+
+      if (overwrite) {
```
Comment (syall) [inline packages/middleware-signing/src/awsAuthConfiguration.ts:208]: nit: `overwrite` reads like the `input.*` parameters would be overwritten, but the code path here is using the original `input.*` parameters.

Comment (kuhe) [reply packages/middleware-signing/src/awsAuthConfiguration.ts:208]: unfortunately this part of the code mutates the config during signing and not during the constructor config resolution

**Review by syall — APPROVED** [review]

**Review by trivikr — APPROVED** [review]

File: packages/middleware-signing/src/awsAuthConfiguration.ts (hunk 2 above)
Comment (kuhe) [reply packages/middleware-signing/src/awsAuthConfiguration.ts]: I changed it to only mutate if authscheme is not sigv4a. Sigv4a doesn't contain a meaningful region so far, so we can skip writing that to the input and leave it to any further resolutions containing a singular region.

**Pull request merged**